community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/convert/commands/ExpressionConverters.scala

```diff
@@ -47,6 +47,8 @@ object ExpressionConverters {
       case e: ast.NotEquals => e.asCommandNotEquals
       case e: ast.RegexMatch => e.asCommandRegex
       case e: ast.In => e.asCommandIn
+      case e: ast.Like => e.asCommandRegex
+      case e: ast.NotLike => e.asCommandNegatedRegex
```

Comment:
Since "a NOT LIKE b" is infix, we supported it as its own expression. Perhaps a rewriter could be considered. Certainly the existing code has "a NOT LIKE b" behaving the same as "NOT(a LIKE b)" with a different parsing chain. They do generate the same AST though. Perhaps that is good enough? See CypherParserTest.scala#316